src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -402,25 +402,30 @@ private void validateCmekKey(FailureCollector failureCollector) {
       try {
         CryptoKeyName cmekKeyName = CryptoKeyName.parse(cmekKey);
         Boolean isServiceAccountFilePath = isServiceAccountFilePath();
-        if (isServiceAccountFilePath != null) {
-          Credentials credentials = getServiceAccount() == null ?
-            null : GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath);
-          Storage storage = GCPUtils.getStorage(getProject(), credentials);
-          Bucket bucket = null;
-          try {
-            bucket = storage.get(getBucket());
-          } catch (StorageException e) {
-            return;
+        Credentials credentials = getServiceAccount() == null ?
+          null : GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath);
+        Storage storage = GCPUtils.getStorage(getProject(), credentials);
+        Bucket bucket = null;
+        try {
+          bucket = storage.get(getBucket());
+        } catch (StorageException e) {
+          /* Ignoring the exception because we don't want the validation to fail if there is an exception getting
+            the bucket information either because the service account used during validation is the CDF service account
+            which is different than the service account that will be used at runtime (the dataproc service account)
+            (assuming the user has auto-detect for the service account) */
+          return;
+        }
+        if (bucket == null) {
+          if (Strings.isNullOrEmpty(location)) {
+            location = "US";
           }
-          if (bucket == null) {
-            String cmekKeyLocation = cmekKeyName.getLocation();
-            if (!cmekKeyLocation.equals(location)) {
-              failureCollector.addFailure(String.format("CMEK key '%s' is in location '%s' while the GCS bucket will " +
-                                                          "be created in location '%s'.", cmekKey,
-                                                        cmekKeyLocation, location)
-                , "Modify the CMEK key or bucket location to be the same")
-                .withConfigProperty(NAME_CMEK_KEY);
-            }
+          String cmekKeyLocation = cmekKeyName.getLocation();
+          if (!cmekKeyLocation.equals(location)) {
```

Comment:
I see a global location for keyrings. Can a global key be used in any location?

What about case sensitivity, is the 'Us' location treated the same as 'US'?

Reply:
Global and multi-region keys are not allowed. Yeah, changed the check to equalsIgnoreCase.

Reply:
'us' is multi-region, does that fail as well?

Reply:
If the location of key is also 'us' then it does not fail